src/renderer/contexts/GlobalNodeState.tsx

```diff
@@ -1136,6 +1139,115 @@ export const GlobalProvider = memo(
+                x: minX,
+                y: minY,
+                width: maxX - minX,
+                height: maxY - minY,
+            };
+        }, [getNodes]);
+
+        const downloadImage = (dataUrl: string, fileName: string) => {
+            const a = document.createElement('a');
+            a.href = dataUrl;
+            a.download = fileName;
+            document.body.appendChild(a);
+            a.click();
+            document.body.removeChild(a);
+        };
+
+        const [viewportExportPadding] = useViewportExportPadding;
+        const exportViewportScreenshot = useCallback(() => {
+            const currentFlowWrapper = reactFlowWrapper.current;
+            if (!(currentFlowWrapper instanceof HTMLElement)) return;
+
+            const oldViewport = currentReactFlowInstance.getViewport();
+
+            const reactFlowViewport = currentFlowWrapper.getBoundingClientRect();
+            const nodesBoundingBox = getNodesBoundingBox();
+
+            if (!nodesBoundingBox) return;
+
+            const paddedBoundingBox = {
+                x: nodesBoundingBox.x - viewportExportPadding,
+                y: nodesBoundingBox.y - viewportExportPadding,
+                width: nodesBoundingBox.width + viewportExportPadding * 2,
+                height: nodesBoundingBox.height + viewportExportPadding * 2,
+            };
+
+            const exportZoom = Math.min(
+                reactFlowViewport.width / paddedBoundingBox.width,
+                reactFlowViewport.height / paddedBoundingBox.height
+            );
+
+            currentReactFlowInstance.setViewport({
+                x: paddedBoundingBox.x * -1 * exportZoom,
+                y: paddedBoundingBox.y * -1 * exportZoom,
+                zoom: exportZoom,
+            });
+
+            // wait for the viewport to be updated
+            setTimeout(() => {
+                toPng(currentFlowWrapper, {
+                    style: {
+                        padding: '0',
+                        margin: '0',
+                        pointerEvents: 'none',
+                    },
+                    pixelRatio: 1 / exportZoom,
+                    width: paddedBoundingBox.width * exportZoom,
+                    height: paddedBoundingBox.height * exportZoom,
+                    filter: (node: unknown) => {
+                        if (
+                            node instanceof HTMLElement &&
+                            (node.classList.contains('react-flow__minimap') ||
+                                node.classList.contains('react-flow__controls'))
+                        ) {
+                            return false;
+                        }
+
+                        return true;
+                    },
+                })
+                    .then((dataUrl: string) => {
+                        currentReactFlowInstance.setViewport(oldViewport);
+
+                        const currentNodename = savePath ? parse(savePath).name : 'Untitled';
```

Comment:
Minor nit but this should probably be called `currentChainName`

Reply:
Whoopsie...
Variable names are hard sometimes